alchemiscale/storage/statestore.py

```diff
@@ -730,6 +745,96 @@ def set_strategy(
         """Set the compute Strategy for the given AlchemicalNetwork."""
         ...
 
+    def register_computeservice(
+        self, compute_service_registration: ComputeServiceRegistration
+    ):
+        """Register a ComputeServiceRegistration uniquely identifying a running
+        ComputeService.
+
+        A ComputeServiceRegistration node is used for CLAIMS relationships on
+        Tasks to avoid collisions in Task execution.
+
+        """
+
+        node = Node(
+            "ComputeServiceRegistration", **compute_service_registration.to_dict()
+        )
+
+        with self.transaction() as tx:
+            tx.merge(
+                node,
+                primary_label="ComputeServiceRegistration",
+                primary_key="identifier",
+            )
+
+        return compute_service_registration.identifier
+
+    def deregister_computeservice(self, compute_service_id: ComputeServiceID):
+        """Remove the registration for the given ComputeServiceID from the
+        state store.
+
+        This wil remove the ComputeServiceRegistration node, and all its CLAIMS
+        relationships to Tasks.
+
+        All Tasks with CLAIMS relationships to the ComputeServiceRegistration
+        and with status `running` will have their status set to `waiting`.
+
+        """
+        q = f"""
+        MATCH (n:ComputeServiceRegistration {{identifier: '{compute_service_id}'}})
+
+        OPTIONAL MATCH (n)-[cl:CLAIMS]->(t:Task {{status: 'running'}})
+        SET t.status = 'waiting'
+
+        DETACH DELETE n
```

Comment:
nitpick but should return `n` from the cypher query and propagate that value if possible.

Reply:
I don't think we can return `n` itself, since we are deleting it. However, I can use the trick we're using below to return the identifier after extracting it with `WITH` prior to the delete.